classes/Group.php

```diff
@@ -95,114 +106,180 @@ public static function clearCachedValues()
+		WHERE cg.`id_group` = '.(int) $this->id.'
 		AND c.`deleted` != 1
-		'.($shop_filtering ? Shop::addSqlRestriction(Shop::SHARE_CUSTOMER) : '').'
+		'.($shopFiltering ? Shop::addSqlRestriction(Shop::SHARE_CUSTOMER) : '').'
 		ORDER BY cg.`id_customer` ASC
-		'.($limit > 0 ? 'LIMIT '.(int)$start.', '.(int)$limit : ''));
+		'.($limit > 0 ? 'LIMIT '.(int) $start.', '.(int) $limit : ''));
     }
 
-    public static function getReduction($id_customer = null)
+    /**
+     * Get Group reduction for Customer
+     *
+     * @param int|null $idCustomer Customer ID
+     *
+     * @return mixed
+     */
+    public static function getReduction($idCustomer = null)
     {
-        if (!isset(self::$cache_reduction['customer'][(int)$id_customer])) {
-            $id_group = $id_customer ? Customer::getDefaultGroupId((int)$id_customer) : (int)Group::getCurrent()->id;
-            self::$cache_reduction['customer'][(int)$id_customer] = Group::getReductionByIdGroup($id_group);
+        if (!isset(self::$cache_reduction['customer'][(int) $idCustomer])) {
+            $idGroup = $idCustomer ? Customer::getDefaultGroupId((int) $idCustomer) : (int) Group::getCurrent()->id;
+            self::$cache_reduction['customer'][(int) $idCustomer] = Group::getReductionByIdGroup($idGroup);
         }
-        return self::$cache_reduction['customer'][(int)$id_customer];
+        return self::$cache_reduction['customer'][(int) $idCustomer];
     }
 
-    public static function getReductionByIdGroup($id_group)
+    /**
+     * Get reduction for Group
+     *
+     * @param int $idGroup Group ID
+     *
+     * @return mixed
+     */
+    public static function getReductionByIdGroup($idGroup)
     {
-        if (!isset(self::$cache_reduction['group'][$id_group])) {
-            self::$cache_reduction['group'][$id_group] = Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue('
+        if (!isset(self::$cache_reduction['group'][$idGroup])) {
+            self::$cache_reduction['group'][$idGroup] = Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue('
 			SELECT `reduction`
 			FROM `'._DB_PREFIX_.'group`
-			WHERE `id_group` = '.(int)$id_group);
+			WHERE `id_group` = '.(int) $idGroup);
         }
-        return self::$cache_reduction['group'][$id_group];
+
+        return self::$cache_reduction['group'][$idGroup];
     }
 
-    public static function getPriceDisplayMethod($id_group)
+    /**
+     * Get price display method for Group
+     * (tax included/excluded)
+     *
+     * @param int $idGroup Group ID
+     *
+     * @return mixed
+     */
+    public static function getPriceDisplayMethod($idGroup)
     {
-        if (!isset(Group::$group_price_display_method[$id_group])) {
-            self::$group_price_display_method[$id_group] = Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue('
+        if (!isset(Group::$group_price_display_method[$idGroup])) {
+            self::$group_price_display_method[$idGroup] = (bool) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue('
```

Comment:
Please do not cast results returned by query execution as it can break existing features.